packages/assets-controllers/src/NftDetectionController.ts

```diff
@@ -546,75 +494,109 @@ export class NftDetectionController extends StaticIntervalPollingController<
+          metadata: { imageOriginal: image_original_url } = {},
+          name,
+          description,
+          attributes,
+          topBid,
+          lastSale,
+          rarityRank,
+          rarityScore,
+          collection,
+        } = nft.token;
+
+        let ignored;
+        /* istanbul ignore else */
+        const { ignoredNfts } = this.#getNftState();
+        if (ignoredNfts.length) {
+          ignored = ignoredNfts.find((c) => {
+            /* istanbul ignore next */
+            return (
+              c.address === toChecksumHexAddress(contract) &&
+              c.tokenId === token_id
+            );
+          });
+        }
+
+        /* istanbul ignore else */
+        if (!ignored) {
           /* istanbul ignore next */
-          return (
-            c.address === toChecksumHexAddress(contract) &&
-            c.tokenId === token_id
+          const nftMetadata: NftMetadata = Object.assign(
+            {},
+            { name },
+            description && { description },
+            image_url && { image: image_url },
+            image_thumbnail_url && { imageThumbnail: image_thumbnail_url },
+            image_original_url && { imageOriginal: image_original_url },
+            kind && { standard: kind.toUpperCase() },
+            lastSale && { lastSale },
+            attributes && { attributes },
+            topBid && { topBid },
+            rarityRank && { rarityRank },
+            rarityScore && { rarityScore },
+            collection && { collection },
           );
-        });
-      }
-
-      /* istanbul ignore else */
-      if (!ignored) {
-        /* istanbul ignore next */
-        const nftMetadata: NftMetadata = Object.assign(
-          {},
-          { name },
-          description && { description },
-          image_url && { image: image_url },
-          image_thumbnail_url && { imageThumbnail: image_thumbnail_url },
-          image_original_url && { imageOriginal: image_original_url },
-          kind && { standard: kind.toUpperCase() },
-          lastSale && { lastSale },
-          attributes && { attributes },
-          topBid && { topBid },
-          rarityRank && { rarityRank },
-          rarityScore && { rarityScore },
-          collection && { collection },
-        );
-
-        await this.#addNft(contract, token_id, {
-          nftMetadata,
-          userAddress,
-          source: Source.Detected,
-          networkClientId: options?.networkClientId,
-        });
-      }
-    });
-    await Promise.all(addNftPromises);
+
+          await this.#addNft(contract, token_id, {
+            nftMetadata,
+            userAddress,
+            source: Source.Detected,
+            networkClientId: options?.networkClientId,
+          });
+        }
+      });
+      await Promise.all(addNftPromises);
+      // update status
+      const isStillFetching = Boolean(resultNftApi.continuation);
+
+      this.#updateNftFetchingProgressStatus(isStillFetching);
```

Comment:
is there a reason this updates on each iteration of the loop vs just setting it to false once after the loop exits?

Reply:
Im showing a spinner at the bottom of the page as long as that value is `true` [here](https://github.com/MetaMask/metamask-extension/pull/24547/files#diff-998630e841bde11aa6060c4b8f4a9fed75944a345243382df69904ead6c4a023R174),

I agree that it is unnecessary to update the state everytime if it already had the same value already!

I think It could be avoided by doing sth similar to this: 
`if (
        isNftFetchingInProgress[userAddress]?.[chainId].isFetchingInProgress !==
        isStillFetching
      ) {
        this.#updateNftFetchingProgressStatus(isStillFetching);
      }`

Reply:
This has been updated in this [commit](https://github.com/MetaMask/core/pull/4281/commits/85d2f52fc7764088392ba9d9e88f21e27b84f7d3) too 🙏